### PR TITLE
ref(*): use the kubernetes client directly (WIP)

### LINKED ISSUE #65: Fix race condition while attempting to stream logs from the builder pod.
https://github.com/deis/builder/blob/daac4b822a94554799c8503e5e7662592492b7f0/rootfs/etc/confd/templates/builder#L139

Is very prone to race conditions. If the builder pod either never makes it to `Running` or transitions through `Running` to `Terminated` we might hang forever.

### LINKED ISSUE #79: use Go templates in manifests
after #72 is done, render templates instead of doing string replaces

### PR DESCRIPTION
**Pull request by arschles**
Closes #79
Fixes #65 

Simplifies dependencies and the code

### CODE REVIEW

File: pkg/gitreceive/build.go [hunk 1]
```diff
@@ -201,53 +152,72 @@ func build(conf *Config, builderKey, gitSha string) error {
...
-	// TODO: use the k8s client and watch the event stream instead (https://github.com/deis/builder/issues/65)
-	for {
-		cmd := kGetCmd(conf.PodNamespace, buildPodName)
-		var out bytes.Buffer
-		cmd.Stdout = &out
-		// ignore errors
-		run(cmd)
-		outStr := string(out.Bytes())
-		if strings.Contains(outStr, "phase: Running") {
+	watcher, err := podsInterface.Watch(labels.Nothing(), fields.OneTermEqualSelector("name", pod.Name), v1Version)
```
Comment (aledbf) [inline pkg/gitreceive/build.go]: @arschles which version of k8s is being targeted in V2?
I ask because this interface is different in v1.2

Comment (arschles) [inline pkg/gitreceive/build.go]: @aledbf In builder, and generally, we're targeting Kubernetes v1.1 for Deis v2 beta.

Sidenote: Some components like [deis/logger](https://github.com/deis/logger) will require [the DaemonSets extension](https://github.com/kubernetes/kubernetes/blob/master/docs/design/daemon.md).

File: pkg/gitreceive/k8s_util.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+	return &api.Pod{
+		TypeMeta: unversioned.TypeMeta{Kind: podKind, APIVersion: v1Version},
+		ObjectMeta: api.ObjectMeta{
+			Name:      name,
+			Namespace: namespace,
+			Labels: map[string]string{
+				"heritage": heritageLabel,
+				"version":  versionLabel,
+			},
+		},
```
Comment (smothiki) [inline pkg/gitreceive/k8s_util.go]: Where are we setting environment variables ?

Comment (arschles) [inline pkg/gitreceive/k8s_util.go]: We aren't yet but it'll have to be done in the below `Spec`. Same in `slugbuilderPod` below

Comment (smothiki) [inline pkg/gitreceive/k8s_util.go]: this is in work in progress  ?

Comment (arschles) [inline pkg/gitreceive/k8s_util.go]: yes, very much so

**Comment by aledbf** [comment]
Closed by #133

**Pull request closed without merging**